rfcs/0002-assetid.md

```diff
@@ -0,0 +1,107 @@
+# Table of Contents
+
+- [Motivation](#motivation)
+- [Guide Level Explanation](#guide-level-explanation)
+- [Reference Level Explanation](#reference-level-explanation)
+- [Drawbacks](#drawbacks)
+- [Rationale and Alternatives](#rationale-and-alternatives)
+- [Prior Art](#prior-art)
+- [Unresolved Questions](#unresolved-questions)
+
+# Basic Info
+[basic]: #basic-info
+
+- Feature Name: asset_id
+- Start Date: 2018-11-20
+- RFC PR: 
+- [Tracking Issue](#tracking-issue): 
+- [Forum Thread](#forum-discussion): 
+
+# Summary
+[summary]: #summary
+
+AssetID is a unified way to reference any loadable asset in Amethyst. It proposes an AssetID enum with variants for v4 UUID (AssetUUID), a file path (PathBuf) or a custom URI format (see [Unresolved Questions](#unresolved-questions)).
+
+Note that this is the first part in a series of RFCs I intend to write that details the technical aspects of my earlier [Asset Pipeline](https://github.com/amethyst/amethyst/issues/875) proposal.
+
+# Motivation
+[motivation]: #motivation
+
+To realize the vision of the Asset Pipeline it will be important to be able to address loadable assets in a more powerful way than with a filesystem path.
+Container formats such as GLTF and FBX can benefit from AssetID where an ID can be generated for each loadable asset within the file as opposed to referencing the entire source file.
+It is also essential that cross-file references use UUID for Amethyst to handle renaming or moving of files without breaking these references.
+
+The primary benefit of AssetID is to make it easier to write tools for Amethyst that handle assets. With a unified system for identifying assets, we can build a unified system for loading assets or metadata about assets which is super useful when creating editors and visualisers.
+
+# Guide-Level Explanation
+[guide-level-explanation]: #guide-level-explanation
+
+AssetID is an identifier for loadable assets. It is a name and enum for any of three variants.
```

Comment:
I thought that we've got rid of the enum here. The ID was supposed to be unique and ideally every asset should have only one.

If you want to preserve the possibility to query by path, i'd say we could add a separate `AssetQuery` type that can hold either an ID or an alias. Dealing with OS paths in general is hard, and i'd leave that out completely if possible.

There is also no word about querying with static type in mind. Can i do `import::<Texture>(assetAlias)`? Or is the returned type always an opaque `dyn Asset`? It's especially noteworthy when you consider potential errors.